scripts/frontmatter-validator/frontmatter-validator.js

```diff
@@ -0,0 +1,93 @@
+const fs = require('fs');
+const ajv = require('ajv');
+const yaml = require('js-yaml');
+const addFormats = require('ajv-formats');
+const ajvErrors = require('ajv-errors');
+
+const schemas = {
+  rule: loadSchema('rule-schema.json'),
+  category: loadSchema('category-schema.json'),
+  top_category: loadSchema('top-category-schema.json'),
+}
+
+const validator = initializeValidator()
+
+function initializeValidator() {
+  const validator = new ajv({ allErrors: true });
+  addFormats(validator);
+  ajvErrors(validator);
+
+  for (const key in schemas) {
+    validator.addSchema(schemas[key], key);
+  }
+
+  return validator
+}
+
+function loadSchema(schemaPath) {
+  return JSON.parse(fs.readFileSync(schemaPath, 'utf8'));
+}
+
+function matchSchema(filePath) {
+  if (filePath.endsWith('rule.md')) {
+    return validator.getSchema('rule')
+  } else if (filePath.indexOf('/categories') !== -1 && !filePath.endsWith('index.md')) {
+    return validator.getSchema('category')
+  } else if (filePath.indexOf('/categories') !== -1 && filePath.endsWith('index.md')) {
+    return validator.getSchema('top_category')
+  }
+}
+
+function validateFrontmatter(filePath) {
+  const fileContents = fs.readFileSync(filePath, 'utf8');
```

Comment:
I think this could use some error handling

What if a file is deleted? Would GitHub still count it as a changed file? If it does this code will try read it

Reply:
Done - error handling has been added. Now the code will check if it exists the file before reading it. 🙂